File: trovaMax.js
function trovaMassimo(array) {
if (array.length === 0) return null;

let max = array[0];
for (let i = 1; i < array.length; i++) {
if (array[i] > max) {
max = array[i];
}
}
return max;
}
let arr2 = [15, 3, 27, 8, 19];
let maxRisultato = trovaMassimo(arr2);
console.log("7) Massimo dell’array =", maxRisultato);

File: trovaMax.py
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let array = [];
let N;
let step = 0;

console.log("Inserisci la dimensione dell'array:");

rl.on('line', (input) => {
  if (step === 0) {
    N = parseInt(input);
    if (isNaN(N) || N <= 0) {
      console.log("Inserisci un numero valido per la dimensione dell'array:");
      return;
    }
    console.log(`Inserisci ${N} elementi:`);
    step++;
  } else if (step > 0 && step <= N) {
    let elemento = parseInt(input);
    if (isNaN(elemento)) {
      console.log("Inserisci un numero valido:");
      return;
    }
    array.push(elemento);
    step++;
    if (step > N) {
      const massimo = trovaMassimo(array, N);
      console.log("Il valore massimo è:", massimo);
      rl.close();
    }
  }
});

function trovaMassimo(arr, n) {
  let massimo = arr[0];
  for (let i = 1; i < n; i++) {
    if (arr[i] > massimo) {
      massimo = arr[i];
    }
  }
  return massimo;
}
